### ISSUE TITLE
Question: What are the network requirements (ports, protocols?)

### ISSUE THREAD

**Issue by stratus-ss** [rospogrigio/localtuya#384]
Thanks for the project and all your effort.

I am trying to setup LocalTuya but it fails to connect to my devices and I believe it is related to the fact that my HomeAssistant Instance is on another VLAN. Even though I have opened up ports 6666-6668 for both TCP and UDP, Home Assistant always marks the device as unavailable.

I can see in the firewall the communication is happening
![image](https://user-images.githubusercontent.com/8162705/110135339-6652ba80-7d9c-11eb-9db3-7228e1a9546b.png)

This makes me wonder if there is some other requirement (such as: are devices required to be in the same broadcast domain?) that could help me enable the functionality?

I have tested the LocalTuya with a vm which I put in the appropriate vlan. Both autodiscovery and manual entry work as described in your documentation.

Any pointers would be appreciated

**Comment by postlund** [comment]
Devices broadcast their presence to UDP port 6666/6667 (that's the ports we listen to at least) so you need to make sure that traffic is forwarded. Port 6668 is the port used to communicate with the device, so you need to keep that open in the firewall 
 as well.

**Comment by stratus-ss** [comment]
Thanks for the answer!

So I have put a `any` `any` rule in the ports between host and vlan but HA still thinks the device is offline.

Is there a way to test this outside of home assistant such as a curl or something? This way I could verify the ports are infact open (even though I see the traffic going through). I strongly suspect its something in home assistant but i need a way to test that

**Comment by postlund** [comment]
Are you sure the rules apply to broadcast traffic? Normally it's only for unicast. I guess you can try to broadcast something with netcat/socat and receive it with netcat on the other network. Some inspiration here (for broadcast):

http://tiebing.blogspot.com/2015/09/using-netcat-to-send-udp-broadcast.html?m=1

**Comment by stratus-ss** [comment]
i'll take a look at that post thanks for the link

As a rule broadcasts don't traverse VLANs or subnets for that matter as they stay in their own broadcast domains. Therefore I am not as worried about having to manually set things up (I already do this with MQTT) but rather, I am trying to determine if broadcast is somehow *required* for the object to be controlled.

If broadcast is required, it is possible that LocalTuya will not work across VLANs. It would be nice to get to the bottom of what is required vs what is nice to have.

**Comment by postlund** [comment]
It is not required. In the latest release I however changed to *only* connect to a device once a broadcast was received from it. The benefits are less re-connect attempts, since Home Assistant doesn't have to try to connect over and over again (the broadcast tells us up and front that the device is ready) and it also speeds up re-connects, as the broadcast is sent as soon as the device can handle a connection. It also simplified code a bit.

This change however broke things for people separating devices and Home Assistant in different VLANs (and you know why). That's why I created #288 that adds periodic re-connects every minute. Once that has been merged and I've made a new release, it will work again. You will however not get discovery working in the config flow (so you will have to enter IP and id manually for all devices) and IP address will not be automatically updated, in case it changes due to DHCP giving it a new IP (which you obviously can solve anyways by assigning it an IP).
